node/src/network.rs

```diff
@@ -250,24 +339,48 @@ pub fn run_network_client(
 /// If the MPC task times out or aborts for any reason, this object must be dropped to ensure
 /// proper cleanup of the associated resources.
 pub struct NetworkTaskChannel {
-    pub task_id: MpcTaskId,
-    my_participant_id: ParticipantId, // for debugging
-    pub participants: Vec<ParticipantId>,
-    connection_versions: Arc<HashMap<ParticipantId, ConnectionVersion>>,
-    sender: Arc<dyn MeshNetworkTransportSender>,
-    receiver: tokio::sync::mpsc::Receiver<MpcPeerMessage>,
+    sender: Arc<NetworkTaskChannelSender>,
+    /// Used for calling receive(&mut self).
+    receiver: tokio::sync::mpsc::UnboundedReceiver<MpcPeerMessage>,
+    /// The set of participants who sent us a Success message; for leader only.
+    successful_participants: HashSet<ParticipantId>,
+    /// Function to clean up relevant data structures in the network transport implementation.
     drop: Option<Box<dyn FnOnce() + Send + Sync>>,
 }
 
-type SendFnForTaskChannel = Arc<
-    dyn Fn(
-            ParticipantId,
-            BatchedMessages,
-            Vec<ParticipantId>,
-        ) -> BoxFuture<'static, anyhow::Result<()>>
-        + Send
-        + Sync,
->;
+/// A subset of the NetworkTaskChannel that doesn't include the mutable parts.
+pub struct NetworkTaskChannelSender {
+    /// The task ID associated with the computation.
+    task_id: MpcTaskId,
+    /// The leader of the computation; there is exactly one leader for each computation.
+    leader: ParticipantId,
+    /// The participant ID of our node.
+    my_participant_id: ParticipantId,
+    /// The participant IDs of all participants in the computation, including our own.
+    participants: Vec<ParticipantId>,
+    /// The version of the connection we have with each other participant.
+    /// This is used to detect dropped connections. See ConnectionVersion for details.
+    connection_versions: HashMap<ParticipantId, ConnectionVersion>,
+    /// The underlying transport layer.
+    transport_sender: Arc<dyn MeshNetworkTransportSender>,
+}
+
+/// A wrapper around NetworkTaskChannel,
+pub struct NetworkTaskChannelWrapper(NetworkTaskChannel);
```

Comment:
Why is this new struct necessary?

Reply:
I'm using a different way to do it; currently it's too ugly.